tests/console/kdump_and_crash.pm

```diff
@@ -15,62 +15,12 @@ use strict;
 use warnings;
 use testapi;
 use utils;
-use power_action_utils 'power_action';
 use kdump_utils;
-use version_utils 'is_sle';
-use registration;
-use Utils::Backends 'is_spvm';
 
 sub run {
-    my ($self) = @_;
-    select_console('root-console');
-
-    # preparation for crash test
-    if (is_sle '15+') {
-        add_suseconnect_product('sle-module-desktop-applications');
-        add_suseconnect_product('sle-module-development-tools');
-    }
-    prepare_for_kdump;
-    activate_kdump;
-
-    # restart to activate kdump
-    power_action('reboot', keepconsole => is_spvm);
-    reconnect_mgmt_console if is_spvm;
-    $self->wait_boot;
-    select_console 'root-console';
-
-    if (check_var('ARCH', 'ppc64le') || check_var('ARCH', 'ppc64')) {
-        if (script_run('kver=$(uname -r); kconfig="/boot/config-$kver"; [ -f $kconfig ] && grep ^CONFIG_RELOCATABLE $kconfig')) {
-            record_soft_failure 'poo#49466 -- No kdump if no CONFIG_RELOCATABLE in kernel config';
-            return 1;
-        }
-    }
-
-    # often kdump could not be enabled: bsc#1022064
-    return 1 unless kdump_is_active;
-    do_kdump;
-    if (get_var('FADUMP')) {
-        reconnect_mgmt_console;
-        assert_screen 'grub2', 180;
-        wait_screen_change { send_key 'ret' };
-    }
-    elsif (is_spvm) {
-        reconnect_mgmt_console;
-    }
-    else {
-        power_action('reboot', observe => 1, keepconsole => 1);
-    }
-    unlock_if_encrypted;
-    # Wait for system's reboot; more time for Hyper-V as it's slow.
-    $self->wait_boot(bootloader_time => check_var('VIRSH_VMM_FAMILY', 'hyperv') ? 200 : undef);
-    select_console 'root-console';
-
-    # all but PPC64LE arch's vmlinux images are gzipped
-    my $suffix = check_var('ARCH', 'ppc64le') ? '' : '.gz';
-    assert_script_run 'find /var/crash/';
-
-    my $crash_cmd = "echo exit | crash `ls -1t /var/crash/*/vmcore | head -n1` /boot/vmlinux-`uname -r`$suffix";
-    validate_script_output "$crash_cmd", sub { m/PANIC:\s([^\s]+)/ }, 600;
+    kdump_utils::configure_service('function');
+    kdump_utils::check_service();
+    kdump_utils::check_function('function');
```

Comment:
To be honest I don't agree with suggested approach. You are basically moving test logic from test file to library and replacing it by generic not-telling-anything functions. I also noticed that it was recently done or other modules as well.
If it some wider plan/project, how it was discusses/designed? Is it somewhere documented?  Are you going to do it for other tests as well? Shouldn't we start to write a new tests with this logic? Would be possible to do it in other way without creating/adding another layer of code? 

Just for the reference:
https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/console/ntp.pm
https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/console/cups.pm

@asmorodskyi @okurz wdyt?

Reply:
we've discussed this for a long time for the service check with develper. you can check out 
more information at here. https://progress.opensuse.org/issues/54995 and https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/7643/.

Reply:
Progress issue https://progress.opensuse.org/issues/54995 doesn't have any technical explanation it contains only the list of packages. Reference do discussion in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/7643 is not  documentation.

Reply:
We don't have formal design documents for the service check project, but we involved many developers, such as okurz,asmorodskyi, jlausuch,Soulofdestiny, cfconrad,rwx788 rodin for the design in the first implementation for http_srv https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/7643, but unfortunately, we didn't involve you to the discuss.

Reply:
It means that your migration team will have to create documentation which will describe what is needed, how it is done and example how to write a new tests which will support migration(to avoid rewriting in the future).

Reply:
yes, we'll create a doc for this.